randomwallpaper@iflow.space/randomWallpaperMenu.js

```diff
@@ -49,6 +52,25 @@ var RandomWallpaperMenu = class {
 
 		this.panelMenu.menu.addMenuItem(new PopupMenu.PopupSeparatorMenuItem());
 
+		// Temporarily pause timer
+		this._pauseTimerItem = new PopupMenu.PopupSwitchMenuItem('Pause timer', false, {});
+		this._pauseTimerItem.sensitive = this.settings.get('auto-fetch', 'boolean');
+		this._pauseTimerItem.setToggleState(this._backendConnection.get('pause-timer', 'boolean'));
+
+		this._pauseTimerItem.connect('toggled', (item, state) => {
+			this._backendConnection.set('pause-timer', 'boolean', state);
+		});
+
+		this.settings.observe('auto-fetch', () => {
+			this._pauseTimerItem.sensitive = this.settings.get('auto-fetch', 'boolean');
+		});
+
+		this._backendConnection.observe('pause-timer', () => {
```

Comment:
By the way, I am not 100% sure right now, but I think the observed value is also passed to the callback as a parameter.
This way, you wouldn't have to query the new value below.

Reply:
Alright, in that case, I'm likely wrong here. :)